src/main/java/uk/gov/hmcts/reform/bulkscanccdeventhandler/services/CaseCreationCallbackHandler.java

```diff
@@ -0,0 +1,101 @@
+package uk.gov.hmcts.reform.bulkscanccdeventhandler.services;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.stereotype.Service;
+import uk.gov.hmcts.reform.bulkscanccdeventhandler.handler.ExceptionRecordEventHandler;
+import uk.gov.hmcts.reform.bulkscanccdeventhandler.handler.model.CaseCreationRequest;
+import uk.gov.hmcts.reform.bulkscanccdeventhandler.handler.model.CaseCreationResult;
+import uk.gov.hmcts.reform.bulkscanccdeventhandler.handler.model.ExceptionRecord;
+import uk.gov.hmcts.reform.ccd.client.model.CaseDetails;
+
+@Service
+public class CaseCreationCallbackHandler {
+
+    private static final Logger log = LoggerFactory.getLogger(CaseCreationCallbackHandler.class);
+
+    private static final String JURISDICTION = "BULKSCAN";
+
+    private final ExceptionRecordEventHandler exceptionRecordEventHandler;
+
+    public CaseCreationCallbackHandler(ExceptionRecordEventHandler exceptionRecordEventHandler) {
+        this.exceptionRecordEventHandler = exceptionRecordEventHandler;
+    }
+
+    public CaseCreationResult handleCaseCreationCallback(
+        CaseDetails exceptionRecordDetails,
+        String eventId,
+        String idamToken,
+        String idamUserId,
+        boolean ignoreWarnings
+    ) {
+        String exceptionRecordId = exceptionRecordDetails.getId() != null
+            ? exceptionRecordDetails.getId().toString()
+            : null;
```

Comment:
is this a valid value?

Reply:
No, it's not. I admit this is a bit weird and in my first approach I validated the ID. Then I realised that the library would validate it, anyway. Another disadvantage would be distributing the validation among many classes/stages of processing, which isn't great.

Reply:
if this happens i.e CCD doesn't returns id of the exception record id then ccd contract is broken and we should probably throw an exception 400 ?

Reply:
400 was always the plan. I'll just bring the initial version, as it's probably less confusing.